src/routes/pull.rs

```diff
@@ -9,8 +9,26 @@ use toml;
 use manifest::Manifest;
 use RAVEN_REPOSITORY_PATH;
 
+#[derive(Debug, FromForm)]
```

Comment:
Your code should follow (at the best you can) Rust's API guidelines:

https://rust-lang-nursery.github.io/api-guidelines/

Even though those are guidelines for the std library, they are also good guidelines for any Rust project.

Especially, you missed the `C-COMMON-TRAITS` guideline.
It's not a real problem as this is a binary and not a library, but you're new to Rust so you might want to have good habits right now even when it's not the most relevant.

Reply:
Could you be a bit more specific for this case please?

Reply:
The given guidelines is a long set of rules to follow in order to write idiomatic Rust code.

You have a checklist (to use before commit) [here](https://rust-lang-nursery.github.io/api-guidelines/checklist.html).

One of this rule is labeled `C-COMMON-TRAITS` (all rules have a label like this, so they can be referenced easily).

More information about this rule can be find [here](https://rust-lang-nursery.github.io/api-guidelines/interoperability.html#types-eagerly-implement-common-traits-c-common-traits).

It basically boils down to "use `#[derive]` on as much standard trait as you can". You'll want to derive Serde traits too, if possible (`Serialize`/`Deserialize`).